tests/vector.rs

```diff
@@ -33,6 +33,109 @@ fn test_from_value() {
     assert_eq!(Vector4::from_value(76.5f64), Vector4::new(76.5f64, 76.5f64, 76.5f64, 76.5f64));
 }
 
+macro_rules! test_op_add {
+    ($VectorN:ident { $($field:ident),+ }, $s:expr, $v:expr) => (
+        // vector + vector ops
+        assert_eq!($v + $v, $VectorN::new($($v.$field + $v.$field),+));
```

Comment:
Once we establish that the by-val implementation works, I'm guessing we could rely on that for the further tests. ie. `assert_eq!(&$v + &$v, $v + $v);`.